lib/cocoapods/installer/podfile_validator.rb

```diff
@@ -0,0 +1,69 @@
+module Pod
+  class Installer
+    # Validate the podfile before installing to catch errors and
+    # problems
+    #
+    class PodfileValidator
```

Comment:
It uses `Downloader.downloader_class_by_key` to verify the strategies this would imply adding `cocoapods-downloader` as a dependency to `core`  or hard coding the downloaders. None of which are appealing solutions imo